src/data_access/repository.py

```diff
@@ -253,27 +253,72 @@ def swap_positions(self, queue_id, pos_1, pos_2):
         self.cursor.execute("""
             UPDATE queue_members
             SET position = CASE position
-                WHEN :pos_1 THEN -1
-                WHEN :pos_2 THEN -2
+                WHEN :pos_1 THEN -:pos_2
+                WHEN :pos_2 THEN -:pos_1
             END
             WHERE queue_id = :queue_id AND position IN (:pos_1, :pos_2)
         """, {
             "queue_id": queue_id,
             "pos_1": pos_1,
             "pos_2": pos_2
         })
+        self.cursor.execute("""
+            UPDATE queue_members
+            SET position = -position
+            WHERE position < 0
+        """)
+
+    def move_queue_member(
+            self,
+            queue_id,
+            user_id_1,
```

Comment:
How are we handling passed user IDs being the same?

Reply:
Due to the UNIQUE constraint we have on the `queue_members` table, if this function receives the same ID for both parameters, it will return the error code below